spec/lib/jobs/ucs-service-profile-discovery-spec.js

```diff
@@ -0,0 +1,146 @@
+// Copyright 2017, EMC, Inc.
+
+'use strict';
+describe('Ucs Discovery Job', function () {
+    var uuid = require('node-uuid'),
+        sandbox = sinon.sandbox.create(),
+        graphId = uuid.v4(),
+        ucsJob,
+        ucsTool,
+        rackmountData,
+        rootData,
+        serverData,
+        waterline = {},
+        Error;
+
+
+    var obm = {
+        config: {
+            uri: "http://10.240.19.220:7080/sys",
+            host: "10.240.19.220",
+            root: "/sys",
+            port: "7080",
+            protocol: "http",
+            username: "ucspe",
+            password: "ucspe",
+            ucs: "10.239.85.153",
+            verifySSL: false
+        },
+        service: "ucs-obm-service"
+    };
+
+    var node = {
+        id: 'abc',
+        type: 'compute',
+        name: 'ls-testServiceProfile',
+        associatedServer: 'sys/test-chassis/test-server',
+        identifiers: [],
+        relations: [],
+        obm: [obm]
+    };
+    
+    before(function() { 
+        helper.setupInjector([
+            helper.require('/lib/jobs/base-job.js'),
+            helper.require('/lib/jobs/ucs-service-profile-discovery.js'),
+            helper.require('/lib/utils/job-utils/ucs-tool.js'),
+            helper.di.simpleWrapper(waterline,'Services.Waterline')
+        ]);
+        waterline.nodes = {
+            create: sandbox.stub().resolves(node),
+            needOne: sandbox.stub().resolves(node),
+            updateOne: sandbox.stub().resolves(node),
+            //needOneById: sandbox.stub().resolves(node)
+        };
+        waterline.obms = {
+            upsertByNode: sandbox.stub().resolves(obm)
+        };
+        Error = helper.injector.get('Errors');
+    });
+    
+    afterEach(function() {
+        sandbox.restore();
```

Comment:
All stubs are created once in `before`, but all are restored after the first test case finishes. So when the 2nd test case begins to run, actually all stubs are not existing, so what's the purpose to create so many stubs in advance?

Meanwhile, I see a lot stubs are never been used, if so, just remove them.